pkg/asset/agent/installconfig.go

```diff
@@ -340,22 +341,45 @@ func (a *OptionalInstallConfig) GetBaremetalHosts() []*baremetal.Host {
 func (a *OptionalInstallConfig) validateBMCConfig(installConfig *types.InstallConfig) field.ErrorList {
 	var allErrs field.ErrorList
 
-	bmcConfigured := false
-	for _, host := range installConfig.Platform.BareMetal.Hosts {
-		if host.BMC.Address == "" {
-			continue
+	if a.isBmcConfigured(installConfig) {
+		switch installConfig.Platform.Name() {
+		case baremetal.Name:
+			{
+				fieldPath := field.NewPath("Platform", "BareMetal")
+				allErrs = append(allErrs, baremetalvalidation.ValidateProvisioningNetworking(installConfig.Platform.BareMetal, installConfig.Networking, fieldPath)...)
+			}
+		case none.Name:
+			{
+				fieldPath := field.NewPath("Platform", "None")
+				allErrs = append(allErrs, nonevalidation.ValidateFencingCredentials(installConfig.Platform.None.FencingCredentials, fieldPath)...)
```

Comment:
I'm not sure if that's the correct location to trigger the validation.
Since this is the `agent` package I suspect it might be the wrong spot and should be somewhere around [here](https://github.com/openshift/installer/blob/main/pkg/asset/installconfig/platformprovisioncheck.go#L175)  :thinking:

Reply:
We will need to call validate, here, but probably not in this PR.
I think your hunch is correct regarding the platformprovisioncheck file.